crates/services/executor/src/executor.rs

```diff
@@ -470,33 +474,36 @@ where
         let source = component.transactions_source;
         let gas_price = component.gas_price;
         let coinbase_contract_id = component.coinbase_contract_id;
-        let mut remaining_gas_limit = block_gas_limit;
         let block_height = *block.header.height();
 
-        if self.relayer.enabled() {
-            self.process_da(&block.header, execution_data)?;
-        }
+        let forced_transactions = if self.relayer.enabled() {
+            self.process_da(&block.header, execution_data)?
+        } else {
+            Vec::with_capacity(0)
+        };
 
         // The block level storage transaction that also contains data from the relayer.
         // Starting from this point, modifications from each thread should be independent
         // and shouldn't touch the same data.
         let mut block_with_relayer_data_transaction = self.block_st_transaction.read_transaction()
-                // Enforces independent changes from each thread.
-                .with_policy(ConflictPolicy::Fail);
+            // Enforces independent changes from each thread.
+            .with_policy(ConflictPolicy::Fail);
 
         // We execute transactions in a single thread right now, but later,
         // we will execute them in parallel with a separate independent storage transaction per thread.
         let mut thread_block_transaction = block_with_relayer_data_transaction
             .read_transaction()
             .with_policy(ConflictPolicy::Overwrite);
 
-        // ALl transactions should be in the `TxSource`.
-        // We use `block.transactions` to store executed transactions.
         debug_assert!(block.transactions.is_empty());
-        let mut iter = source.next(remaining_gas_limit).into_iter().peekable();
+        // initiate transaction stream with relayed (forced) transactions first,
+        // and pull the rest from the TxSource (txpool) if there is remaining blockspace available.
+        // We use `block.transactions` to store executed transactions.
+        let relayed_tx_iter = forced_transactions.into_iter().peekable();
```

Comment:
[nit] maybe we should just move this line down to the for loop for this iterator? seems odd to have it way up here still.